test_scons_local.sh

```diff
@@ -66,7 +66,10 @@ set +e
 command -v clang-format >> /dev/null 2>&1
 if [ "${PIPESTATUS[0]}" -eq 0 ]; then
     set -e
+    directory=$(pwd)
+    ln -s "${directory}" "${directory}/test/tool/site_scons"
     scons -C test/tool -f SConstruct
+    rm "${directory}/test/tool/site_scons"
```

Comment:
Would this be better as an _exit trap_ given:

https://github.com/daos-stack/scons_local/blob/07e87446f6e57ddfa6bb03d0c6907b66bb3f49c4/test_scons_local.sh#L22
will mean the `rm` won't be done if `scons` returns with an error.